src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -268,16 +282,21 @@ StartRemoteTransactionCommit(MultiConnection *connection)
 	}
 	else if (transaction->transactionState == REMOTE_TRANS_PREPARED)
 	{
-		/* commit the prepared transaction */
-		StringInfoData command;
-
-		initStringInfo(&command);
-		appendStringInfo(&command, "COMMIT PREPARED %s",
-						 quote_literal_cstr(transaction->preparedName));
+		/*
+		 * Commit the prepared transaction.
+		 *
+		 * We need to allocate 148 bytes for command buffer (including '\0'):
+		 *  - len("COMMIT PREPARED ") = 16
+		 *  - maximum quoted length of transaction->preparedName = 2 * 64 + 3 = 131
```

Comment:
what is this length calculation based on?

If we want the general postgres limit it appears to be 200 bytes:
https://www.postgresql.org/docs/current/sql-prepare-transaction.html

Reply:
I thought that length of `preparedname` could at most be NAMEDATALEN(64) and quoting a string would mean 2*X+3 bytes at the worst case.

```c
char *
quote_literal_cstr(const char *rawstr)
{
....
....
	/* We make a worst-case result area; wasting a little space is OK */
	result = palloc(len * 2 + 3 + 1);
....
....
```

However, now I see that NAMEDATALEN can be configured before initializing the database directory, so I guess I should replace 64 with 200 in the calculation.